client/src/ProtectedRoute.js

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { Redirect, Route } from "react-router-dom";
+
+import auth from './components/auth/auth'
+
+export default function PrivateRoute({ children, ...rest }) {
+  return (
+    <Route
+      {...rest}
+      render={({ location }) =>
+        auth.isAuthenticated() ? (
+          children
+        ) : (
+            <Redirect
+              to={{
+                pathname: "/",
+                state: { from: location }
+              }}
+            />
+          )
+      }
+    />
+  );
+}
```

Comment:
```suggestion
export default function PrivateRoute({ children, ...rest }) {
  return auth.isAuthenticated() ? (
    <Route {...rest}>{children}</Route>
  ) : (
    <Redirect
      to={{
        pathname: "/"
      }}
    />
  );
}

```

Reply:
I noticed this still wasn't working properly, since it wouldn't redirect to `/` when attempting to access `/newgame` directly. To use a <Redirect> component, you either return it on a render() method, as I did above, or you place it directly inside a <Switch> and give it a `from` prop indicating which route should be redirected to `to`